mlx/linalg.cpp

```diff
@@ -292,4 +292,41 @@ array cholesky(
       {a});
 }
 
+array pinv(const array& a, StreamOrDevice s /* = {} */) {
+  if (a.dtype() != float32) {
+    std::ostringstream msg;
+    msg << "[linalg::pinv] Arrays must type float32. Received array "
+        << "with type " << a.dtype() << ".";
+    throw std::invalid_argument(msg.str());
+  }
+  if (a.ndim() < 2) {
+    std::ostringstream msg;
+    msg << "[linalg::pinv] Arrays must have >= 2 dimensions. Received array "
+           "with "
+        << a.ndim() << " dimensions.";
+    throw std::invalid_argument(msg.str());
+  }
+
+  const auto m = a.shape(-2);
+  const auto n = a.shape(-1);
+  const auto k = std::min(m, n);
+  const auto rank = a.ndim();
```

Comment:
`rank` is not used anywhere .

Reply:
Removed ✅ - in new commit 587a30f6; [diff here](https://github.com/ml-explore/mlx/pull/875/commits/587a30f6bd32a12a0f58ed660fc7a5144820c3b1#diff-9f87ac95c2fab89844d98cd9bd7727e2e7a872880ea49db8baf529bb43bdfcddL313)